security/tpm/test/_tpm_test.js

```diff
@@ -92,6 +149,29 @@ describe('tpm', function () {
         done();
       });
     });
+
+    it('creates a storage root key if one does not exist', function(done) {
+      /*Tests_SRS_NODE_TPM_SECURITY_CLIENT_06_018: [If the storage root key does NOT exist, a new key will be created.] */
+      var tpm = new tss.Tpm();
+      var connectStub = sinon.stub(tpm, 'connect');
+      connectStub.callsArgWith(0);
+      var client = new TpmSecurityClient(undefined, tpm);
+      var readPublicStub = sinon.stub(tpm, 'ReadPublic');
+      var lastResponseStub = sinon.stub(tpm, 'getLastResponseCode');
+      lastResponseStub.returns(tss.TPM_RC.SUCCESS);
+      lastResponseStub.onCall(1).returns(tss.TPM_RC.BAD_TAG);
+      var createPrimaryStub = sinon.stub(tpm, 'CreatePrimary');
+      createPrimaryStub.callsArgWith(5, fakeCreatedSrkKey);
+      var evictControlStub = sinon.stub(tpm, 'EvictControl').callsArg(3);
+      var flushContextStub = sinon.stub(tpm, 'FlushContext').callsArg(1);
+      readPublicStub.withArgs(TpmSecurityClient._ekPersistentHandle).callsArgWith(1, fakeEkKey);
+      readPublicStub.withArgs(TpmSecurityClient._srkPersistentHandle).callsArgWith(1, fakeSrkKey);
+      readPublicStub.withArgs(TpmSecurityClient._idKeyPersistentHandle).callsArgWith(1, fakeIdKey);
+      client.getStorageRootKey((err, localSrk) => {
+        assert.deepEqual(localSrk, fakeCreatedSrkKey.outPublic.asTpm2B(), 'Invalid endorsement key returned.');
```

Comment:
>endorsement [](start = 83, length = 11)

typo: storage root key?

Reply:
Thank you.

---
In reply to: [167385969](https://github.com/Azure/azure-iot-sdk-node/pull/225#discussion_r167385969) [](ancestors = 167385969)